### PR TITLE
generalized gamma distribution

### PR DESCRIPTION
**Pull request by jcalifornia**
Generalized gamma distribution addressing #1081 

Uses wikipedia parameterization: https://en.wikipedia.org/wiki/Generalized_gamma_distribution

### CODE REVIEW

File: tensorflow_probability/python/distributions/generalized_gamma.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+from tensorflow_probability.python.internal import reparameterization
+from tensorflow_probability.python.internal import tensor_util
+from tensorflow_probability.python.bijectors import softplus as softplus_bijector
+
+
+__all__ = [
+  'GeneralizedGamma',
+]
+
+tfd = tfp.distributions
```
Comment (brianwa84) [inline tensorflow_probability/python/distributions/generalized_gamma.py:40]: We can't use these in here, need to instead use direct imports of the modules in question.

**Comment by joshchang** [comment]
Not sure what the process is for you guys - do you want me to make these updates or are you guys going to do it internally on your team?

**Comment by brianwa84** [comment]
For comments on the PR we'd ask you to make the updates. Also take a look at the linter outputs. We sometimes make minor tweaks when we pull things in.

File: tensorflow_probability/python/distributions/generalized_gamma.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    """
+    gamma_samples = tf.random.gamma(
+      shape=[n],
+      alpha=self.shape/self.exponent,
+      beta=1.,
+      dtype=self.dtype,
+      seed=seed
+    )
+    ggamma_samples = (
+      self.scale*tf.math.exp(tf.math.log(gamma_samples)/self.exponent)
```
[the remark this answers is not in the document]
Comment (joshchang) [reply tensorflow_probability/python/distributions/generalized_gamma.py:195]: what do you think the third parameter should be called>

**Pull request closed without merging**